### ISSUE TITLE
getInitialData is lost when using Vue.extend

### ISSUE THREAD

**Issue by IlyaSemenov** [ream/ream#117]
The following code doesn't print `foo`:

```vue
<template>
  <div>{{ foo }}</div>
</template>

<script>
export default Vue.extend({
	getInitialData () {
		return {
			foo: 123
		}
	},
})
</script>
```

however it starts to work if `Vue.extend` if removed. This makes it hard to use Ream in Typescript (which relies on `Vue.extend` to infer typings).

I belive the reason is here: https://github.com/ream/ream/blob/326e41faec7406cad75c5195b3b7959297a78c64/app/server-entry.js#L65
it should not only allow support raw component options but also ready-made Vue subclasses.

**Comment by egoist** [comment]
maybe we can fix this and #101 at the same time by checking `Component.options.getInitialData`

**Comment by IlyaSemenov** [comment]
I think it's worth looking at #101 at the same time, as the solution could be shared (or could be not... not sure if it's possible to somehow force and cache component 'compilation').

**Comment by IlyaSemenov** [comment]
wow. what a timing lol.

**Comment by IlyaSemenov** [comment]
Fixed in 1807da00d9a510a25b1556da0a3cf39350d1f01a (ream 3.2.10). I referred to #18 in the commit message by mistake.

#101 can't be fixed universally with the same approach. This is because in Vue, options are merged late (either when component is instantiated, or during `Vue.extend` call). Even if we add proper merge policy for `getInitialData` property, it will not work with 'simple' data object exports, as we use them before they are instantiated (and thus merged).

I believe #101 will need 'manual' recursive lookup through mixins.
